packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -636,6 +636,93 @@ void main() {
     await testPositioningDownThenUp(tester, TextDirection.rtl, Alignment.bottomCenter, TextDirection.rtl, const Rect.fromLTWH(450.0, 500.0, 0.0, 0.0));
   });
 
+  testWidgets('PopupMenu positioning inside nested Overlay', (WidgetTester tester) async {
+    final Key buttonKey = UniqueKey();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          appBar: AppBar(title: const Text('Example')),
+          body: Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: Overlay(
+              initialEntries: [
+                OverlayEntry(
+                  builder: (_) => Center(
+                    child: PopupMenuButton<int>(
+                      key: buttonKey,
+                      itemBuilder: (_) => <PopupMenuItem<int>>[
+                        const PopupMenuItem<int>(value: 1, child: Text('Item 1')),
+                        const PopupMenuItem<int>(value: 2, child: Text('Item 2')),
+                      ],
+                      child: const Text('Show Menu'),
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final Finder buttonFinder = find.byKey(buttonKey);
+    final Finder popupFinder = find.byWidgetPredicate(
+      (Widget w) => '${w.runtimeType}' == '_PopupMenu<int>',
+    );
```

Comment:
Instead of relying on the private type, could you just check that the position of the first entry in the Popup menu is correct?

Reply:
First entry has a little shift from the top edge. It would be better to use ``find.bySemanticsLabel('Popup menu')``.